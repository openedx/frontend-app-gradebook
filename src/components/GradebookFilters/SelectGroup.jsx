/* eslint-disable react/sort-comp, react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';

import { Form } from '@edx/paragon';

const SelectGroup = ({
  id,
  label,
  value,
  onChange,
  disabled,
  options,
}) => (
  <div className="student-filters">
    <Form.Group controlId={id}>
      <Form.Label>{label}</Form.Label>
      <Form.Control as="select" {...{ value, onChange, disabled }}>
        {options}
      </Form.Control>
    </Form.Group>
  </div>
);
SelectGroup.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.node.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  options: PropTypes.arrayOf(PropTypes.node).isRequired,
};
SelectGroup.defaultProps = {
  disabled: false,
};

export default SelectGroup;
