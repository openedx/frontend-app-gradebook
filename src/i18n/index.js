import { messages as footerMessages } from '@edx/frontend-component-footer';
import { messages as headerMessages } from '@edx/frontend-component-header';
import { messages as paragonMessages } from '@edx/paragon';

import arMessages from './messages/ar.json';
import deMessages from './messages/de.json';
import es419Messages from './messages/es_419.json';
import faIRMessages from './messages/fa_IR.json';
import frMessages from './messages/fr.json';
import frCAMessages from './messages/fr_CA.json';
import hiMessages from './messages/hi.json';
import itMessages from './messages/it.json';
import ptMessages from './messages/pt.json';
import ruMessages from './messages/ru.json';
import ukMessages from './messages/uk.json';
import zhcnMessages from './messages/zh_CN.json';
// no need to import en messages-- they are in the defaultMessage field

const appMessages = {
  ar: arMessages,
  'es-419': es419Messages,
  'fa-ir': faIRMessages,
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
};

export default [
  paragonMessages,
  footerMessages,
  headerMessages,
  appMessages,
];
