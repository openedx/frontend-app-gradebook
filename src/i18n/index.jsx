import arMessages from './messages/ar.json';
// no need to import en messages-- they are in the defaultMessage field
import es419Messages from './messages/es_419.json';
import frMessages from './messages/fr.json';
import zhcnMessages from './messages/zh_CN.json';

const messages = {
  ar: arMessages,
  'es-419': es419Messages,
  fr: frMessages,
  'zh-cn': zhcnMessages,
};

export default messages;
