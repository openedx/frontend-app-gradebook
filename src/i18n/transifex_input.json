{
  "gradebook.BulkManagementTab.csvUploadLabel": "Upload Grade CSV",
  "gradebook.BulkManagementTab.heading": "Use this feature by downloading a CSV for bulk management, overriding grades locally, and coming back here to upload.",
  "gradebook.BulkManagementTab.hint1": "Results appear in the table below.",
  "gradebook.BulkManagementTab.hint2": "Grade processing may take a few seconds.",
  "gradebook.BulkManagementTab.importBtnText": "Import Grades",
  "gradebook.BulkManagementTab.successDialog": "CSV processing. File uploads may take several minutes to complete."
}