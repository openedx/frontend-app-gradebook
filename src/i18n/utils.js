import { getLocale, isRtl } from '@edx/frontend-platform/i18n';

export const getLocalizedSlash = () => {
  // For fractional grades
  // if we are in a LTR language, we want to use a forward slash.
  // If we are in a RTL language, we want to use a backslash instead
  if (isRtl(getLocale())) {
    return '\\';
  }
  return '/';
};

export const getLocalizedPercentSign = () => {
  // LTR languages put the percent to the right of a number.
  // RTL languages put the percent sign to the left of the number.
  // We can place a non-printing unicode right-to-left marker next to the percent
  // sign to make it print to the left of the number if we are currently in a LTR language
  if (isRtl(getLocale())) {
    return '\u200f%';
  }
  return '%';
};
