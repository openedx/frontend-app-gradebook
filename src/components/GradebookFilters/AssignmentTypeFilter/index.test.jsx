import React from 'react';
import { shallow } from 'enzyme';
import { useIntl } from '@edx/frontend-platform/i18n';

import SelectGroup from '../SelectGroup';
import useAssignmentFilterTypeData from './hooks';
import AssignmentFilterType from '.';

jest.mock('../SelectGroup', () => 'SelectGroup');
jest.mock('./hooks', () => ({ __esModule: true, default: jest.fn() }));

const handleChange = jest.fn();
const testType = 'test-type';
const assignmentTypes = [testType, 'type1', 'type2', 'type3'];
useAssignmentFilterTypeData.mockReturnValue({
  handleChange,
  selectedAssignmentType: testType,
  assignmentTypes,
  isDisabled: true,
});

const updateQueryParams = jest.fn();

let el;
describe('AssignmentFilterType component', () => {
  beforeAll(() => {
    el = shallow(<AssignmentFilterType updateQueryParams={updateQueryParams} />);
  });
  describe('behavior', () => {
    it('initializes hooks', () => {
      expect(useAssignmentFilterTypeData).toHaveBeenCalledWith({ updateQueryParams });
      expect(useIntl).toHaveBeenCalledWith();
    });
  });
  describe('render', () => {
    test('snapshot', () => {
      expect(el).toMatchSnapshot();
    });
    test('filter options', () => {
      const { options } = el.find(SelectGroup).props();
      expect(options.length).toEqual(5);
      const optionProps = options[1].props;
      expect(optionProps.value).toEqual(assignmentTypes[0]);
      expect(optionProps.children).toEqual(testType);
    });
  });
});
