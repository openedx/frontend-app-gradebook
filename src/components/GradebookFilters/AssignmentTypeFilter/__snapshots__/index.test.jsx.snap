// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignmentFilterType component render snapshot 1`] = `
<div
  className="student-filters"
>
  <SelectGroup
    disabled={true}
    id="assignment-types"
    label="Assignment Types"
    onChange={[MockFunction]}
    options={
      Array [
        <option
          value=""
        >
          All
        </option>,
        <option
          value="test-type"
        >
          test-type
        </option>,
        <option
          value="type1"
        >
          type1
        </option>,
        <option
          value="type2"
        >
          type2
        </option>,
        <option
          value="type3"
        >
          type3
        </option>,
      ]
    }
    value="test-type"
  />
</div>
`;
